ResearchKit/Common/ORKBaseGraphView.m

```diff
@@ -0,0 +1,870 @@
+                    self.minimumValue = num;
+                }
+            }
+        }
+        
+    }
+    
+    // Maximum
+    if ([self.dataSource respondsToSelector:@selector(maximumValueForGraphView:)]) {
+        self.maximumValue = [self.dataSource maximumValueForGraphView:self];
+    } else {
+        if (self.dataPoints.count) {
+            self.maximumValue = ((ORKRangePoint *)self.dataPoints[0]).maximumValue;
+            
+            for (NSUInteger i=1; i<self.dataPoints.count; i++) {
+                CGFloat num = ((ORKRangePoint *)self.dataPoints[i]).maximumValue;
+                if (((num != NSNotFound) && (num > self.maximumValue)) || (self.maximumValue == NSNotFound)) {
+                    self.maximumValue = num;
+                }
+            }
+        }
+    }
+}
+
+- (CGFloat)snappedXPosition:(CGFloat)xPosition {
+    CGFloat widthBetweenPoints = CGRectGetWidth(self.plotsView.frame)/self.xAxisPoints.count;
+    NSUInteger positionIndex;
+    for (positionIndex = 0; positionIndex<self.xAxisPoints.count; positionIndex++) {
+        
+        CGFloat dataPointVal = ((ORKRangePoint *)self.dataPoints[positionIndex]).maximumValue;
+        
+        if (dataPointVal != NSNotFound) {
+            CGFloat num = [self.xAxisPoints[positionIndex] floatValue];
+            
+            if (fabs(num - xPosition) < (widthBetweenPoints * SnappingClosenessFactor)) {
+                xPosition = num;
+            }
+        }
+    }
+    return xPosition;
+}
+
+- (CGFloat)valueForCanvasXPosition:(CGFloat)xPosition {
+    BOOL snapped = [self.xAxisPoints containsObject:@(xPosition)];
+    CGFloat value = NSNotFound;
+    NSUInteger positionIndex = 0;
+    
+    if (snapped) {
+        for (positionIndex = 0; positionIndex<self.xAxisPoints.count-1; positionIndex++) {
+            CGFloat xAxisPointVal = [self.xAxisPoints[positionIndex] floatValue];
+            if (xAxisPointVal == xPosition) {
+                break;
+            }
+        }
+        value = ((ORKRangePoint *)self.dataPoints[positionIndex]).maximumValue;
+    }
+    
+    return value;
+}
+
+- (NSInteger)yAxisPositionIndexForXPosition:(CGFloat)xPosition {
+    NSUInteger positionIndex = 0;
+    for (positionIndex = 0; positionIndex<self.xAxisPoints.count-1; positionIndex++) {
+        CGFloat xAxisPointVal = [self.xAxisPoints[positionIndex] floatValue];
+        if (xAxisPointVal == xPosition) {
+            break;
+        }
+    }
+    return positionIndex;
+}
+
+- (CAShapeLayer *)plotLineLayerForPlotIndex:(NSInteger)plotIndex withPath:(CGPathRef)path {
+    CAShapeLayer *layer = [CAShapeLayer layer];
+    layer.path = path;
+    layer.fillColor = [UIColor clearColor].CGColor;
+    layer.strokeColor = (plotIndex == 0) ? self.tintColor.CGColor : self.referenceLineColor.CGColor;
+    layer.lineJoin = kCALineJoinRound;
+    layer.lineCap = kCALineCapRound;
+    layer.opacity = 0.4;
```

Comment:
I think this line is an error. Where did it come from?

The line opacity was 1.0 before the refactor.